scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+    
+    Returns:
+        (str): Triplet name.
+               Eg: "x64-windows-static".:
+    """
+    triplet_name = [arch]
+    if is_windows_os():
+        triplet_name.append('windows')
+        # For windows, default is to build dynamic. Hence we specify static.
+        # For mac/linux, default is to build static libraries
+        triplet_name.append('static')
+    elif is_mac_os():
+        triplet_name.append('osx')
+    elif is_linux_os():
+        triplet_name.append('linux')
+    else:
+        raise ValueError('Unsupported operating system \
+                          We currently support windows, mac, linux')
+
+    return '-'.join(triplet_name)
+
+
+def install_packages_with_vcpkg(arch, build_vcpkg=True):
+    """Install packages with vcpkg.
+
+    This does the following,
+        - initialize/update the vcpkg git submodule
+        - build vcpkg executable
+        - install packages via vcpkg.
+    Args:
+        arch (str): Architecture (eg: 'x86', 'x64').
+        install_vcpkg (bool): Build/Install vcpkg before building packages?
+    """
+    # vcpkg is a submodule of the main cpp repo
+    # Ensure that the submodule is initialized and updated
+    run_command(['git', 'submodule', 'init'])
+    run_command(['git', 'submodule', 'update'])
+
+    if build_vcpkg:
+        # build and install vcpkg
+        script_absolute_path = get_vcpkg_installation_script_path()
+        run_command([script_absolute_path])
+    else:
+        print("Skipping building/installing vcpkg executable")
+
+    # for each desktop platform, there is an existing vcpkg response file 
+    # in the repo (external/vcpkg_<triplet>_response_file.txt) defined for each target triplet
+    vcpkg_triplet = get_vcpkg_triplet(arch)
+    print('vcpkg triplet auto determined: {0}'.format(vcpkg_triplet))
+    vcpkg_response_file_path = get_vcpkg_response_file_path(vcpkg_triplet)
+    vcpkg_executable_file_path = get_vcpkg_executable_file_path()
+
+    run_command([vcpkg_executable_file_path, 'install', '@' + vcpkg_response_file_path, '--disable-metrics'])
+
+
+def step_install_system_packages(args):
+    """ Install packages (dependencies) with system package manager.
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments.
+    """
+    system_packages_to_install = []
+    # Install ninja and/or ccache if needed
+    # check if ninja is already installed before attempting to install it
+    if args.generator == 'Ninja':
+        if is_tool_installed('ninja'):
+            print('Ninja requested but is already installed. Skipping installation...')
+        elif is_linux_os():
+            system_packages_to_install.append('ninja-build')
+        elif is_mac_os:
+            system_packages_to_install.append('ninja')   
+
+    # ccache is supported only on mac and linux and check if it is already installed
+    if args.ccache and (is_linux_os() | is_mac_os()) and not is_tool_installed('ccache'):
```

Comment:
A couple of things here. First, I'm assuming the '|' is supposed to be 'or'?

Second, I might be misreading this since it's late on a Friday, but let's say
a = args.ccache
b = is_linux_os() or is_mac_os()
c = is_tool_installed('ccache')

Then the outer conditional is "if a and b and not c"

The first inner conditional is "if c", which is always false at this point.

The second inner conditional is "if b", which is always true at this point.

Reply:
My bad. I folded the nested conditions into one condition and didnt remove the old blocks. Thanks for catching this!